src/languages/smtlib2/poly/ast.ml

```diff
@@ -172,7 +172,7 @@ module type Statement = sig
 
   (** Checking for satisfiablity *)
 
-  val check_sat : ?loc:location -> term list -> t
+  val check_sat : ?loc:location -> ?name:id -> term list -> t
```

Comment:
I'm not fond of adding an effectively unused parameter here. Instead of trying to use the same `check_sat` function for both `smtlib` and `ae`, how about `ae` uses a new function, e.g. `prove_sat` that would take the `.name` argument ?

Reply:
Ok :)